File: java/patternsample/src/main/java/pattern/sample/patternuse01/building/Barrack.java
package pattern.sample.patternuse01.building;

import pattern.sample.patternuse01.code.UnitType;
import pattern.sample.patternuse01.unit.Empty;
import pattern.sample.patternuse01.unit.Marine;
import pattern.sample.patternuse01.unit.Medic;
import pattern.sample.patternuse01.unit.inf.Unit;

public class Barrack {

    public static Unit createUnit(UnitType unitType){
        switch (unitType){
            case MEDIC:
                return new Medic();
            case MARINE:
                return new Marine();
            default:
                return new Empty();
        }

    }
}

File: kotlin_basic/src/main/kotlin/bong/lines/patterns/abstractfactory/building/Barrack.kt
package bong.lines.patterns.abstractfactory.building

import bong.lines.patterns.abstractfactory.code.UnitType
import bong.lines.patterns.abstractfactory.unit.Marine
import bong.lines.patterns.abstractfactory.unit.Medic
import bong.lines.patterns.abstractfactory.unit.inf.Unit

class Barrack {
    fun create(unitType: UnitType) : Unit {
        return when(unitType){
            UnitType.MARINE -> Marine()
            UnitType.MEDIC -> Medic()
        }
    }
}